contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,75 @@
+# REST Architectural Style
+
+Author: [Wen Xin](https://github.com/wenmogu)
+
+## Definition
+Representational State Transfer (REST) is the architecture style of the World Wide Web we are using today.
+
+## Why learn REST 
+It is the set of guiding principles of the development of the Web (i.e. how the Web should be developed such that it can be efficient and widely applicable). It also serves as the model for Web protocols e.g. HTTP and the guiding principles for [RESTful API](https://openclassrooms.com/courses/build-your-web-projects-with-rest-apis/examples-of-rest-apis), the type of API that most of the companies use to provide their web services. Hence, understanding REST will give us an idea of the rationale behind the design of the Web and let us have a better understanding of the key properties of RESTful API. The knowledge about REST would come in handy especially when we need to use other people’s RESTful APIs and implement our own RESTful APIs. 
+
+## <a name="meaning-of-rest"></a>Meaning of REST
+REST stands for “Representational State Transfer”. Before we understand the term “representational state”, we have to understand the term “resource”. In the context of REST, the term “resource” is used as the way to organize and represent data/information. Roy Fielding, the originator of REST, states in his paper that “any information that can be named can be a resource”, and the identification of resources is done by using unique identifiers of the resources i.e. unique name of the resources. The information represented by one resource can change at any time, but the name of the resource i.e. the identifier of the resource has to remain the same for identification purposes. For example, the student list of a computer science course with course code CS3281 can be named as “CS3281StudentList”, and hence can be a resource. The student list might change over time, e.g. from those who enrolled in Year 2017 to those who enrolled in Year 2018, but the identifier “CS3281StudentList” always refers to the student list regardless of its state/value. 
```

Comment:
When you introduce a new term, you can use _italics_. Alternatively, use single quotes, but not double quotes.